src/source/Ice/SocketConnection.c

```diff
@@ -236,13 +236,17 @@ STATUS socketConnectionClosed(PSocketConnection pSocketConnection)
 
     CHK(pSocketConnection != NULL, STATUS_NULL_ARG);
     CHK(!ATOMIC_LOAD_BOOL(&pSocketConnection->connectionClosed), retStatus);
-    MUTEX_LOCK(pSocketConnection->lock);
+    if (IS_VALID_MUTEX_VALUE(pSocketConnection->lock)) {
```

Comment:
There are two cases when this could happen.

1) If the construction of the object is incomplete - this is not the case as we check for the mutex successful creation in LN21 so we teardown on error. 
2) If your code accesses an already collected/freed SocketConnection object

You should identify the root cause of the second issue in your application and understand why something is attempting to access a freed object.

Reply:
If it's case 2, it will goto CleanUp when do `CHK(pSocketConnection != NULL, STATUS_NULL_ARG);` at LN237. I was thinking there might be one scenario:
1. At https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/ec0d7844aaa81f062218c63d829d9fe1d64074be/src/source/Ice/SocketConnection.c#L70, we will free lock
2. If https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/ec0d7844aaa81f062218c63d829d9fe1d64074be/src/source/Ice/SocketConnection.c#L77 failed, we will skip `MEMFREE(pSocketConnection);` and `*ppSocketConnection = NULL;`

This might cause `lock` is invalid but `pSocketConnection != NULL` ?

Reply:
Oh wow.. Well, on the teardown path we won't be calling this over again to hit this case. However, this is a serious issue you've spotted. None of our "destructors" should do these jumps. The destruction is a best effort approach as the destructor might be working on a partially assembled object. LN 77 should be fixed ASAP. Will send a fix immediately

Reply:
Please check this out: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/1001